src/hotspot/share/c1/c1_LIR.hpp

```diff
@@ -198,14 +198,14 @@ class LIR_Const: public LIR_OprPtr {
 class LIR_Opr {
  public:
   // value structure:
-  //     data       opr-type opr-kind
-  // +--------------+-------+-------+
-  // [max...........|7 6 5 4|3 2 1 0]
+  //     data      opr-type  opr-kind
+  // +-----------+----------+-------+
+  // [max........|6  5  4  3|2  1  0]
```

Comment:
I don't think your change is right. The code below indicates there are 3 bits for kind. So I think the original should actually look like this:

```
//     data       opr-type opr-kind
// +--------------+-------+--------+--+
// [max...........|7 6 5 4| 3 2 1  | 0]

```

Reply:
According to the definitions of `Opr_kind` and `kind_mask` below, it can be seen that `opr-kind` uses the lowest 3 bits, and if the lowest bit is 0, it means that this is a pointer.

Reply:
I was reading:
```
enum OprBits {
      pointer_bits   = 1
    , kind_bits      = 3
    , type_bits      = 4
    , size_bits      = 2
    , destroys_bits  = 1
```
as a bitfield description so 1 pointer bit followed by 3 kind bits, followed by 4 type bits etc. But I agree the later mask value only does a shift of 3 not 4.

Also the code indicates there are a lot more non-data bits before we get to the data so the diagram is incomplete in other ways.

Reply:
I don't think `pointer_bits` should be added to `non_data_bits`:

```cpp
    , non_data_bits  = pointer_bits + kind_bits + type_bits + size_bits + destroys_bits + virtual_bits
                       + is_xmm_bits + last_use_bits + is_fpu_stack_offset_bits
    , data_bits      = BitsPerInt - non_data_bits
    , reg_bits       = data_bits / 2      // for two registers in one value encoding
```

Reply:
I agree.  This was changed as part of JDK-8261235.
@chhagedorn, I think there must be a different issue with vreg_max in JDK-8261235.  I wounder if it should be using reg_bits instead of data_bits.